<a href="https://colab.research.google.com/github/Rohit131313/Amazon-Mobile-Scraping-Analysis/blob/main/Amazon_Mobile_Dataset_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Mobile Analysis (Amazon)

In [ ]:
import pandas as pd
import os

In [ ]:
df=pd.read_csv('./amazon_data.csv')

In [ ]:
df

title    price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32,999.   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15,990.   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8,499.   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9,899.   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14,499.   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11,999.   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10,999.   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5,299.   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12,999.   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14,999.   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25,999.   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6,799.   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10,999.   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12,999.   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7,299.   
15  IKALL K510 Smartphone (5" Display, 2GB, 32GB, ...   4,999.   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12,999.   
17  Motorola E13 (Aurora Green, 8GB RAM, 128GB Sto...   8,990.   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12,599.   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12,999.   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11,999.   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19,999.   

                rating         reviews availability  
0   4.3 out of 5 stars   1,095 ratings     In stock  
1   4.0 out of 5 stars  12,391 ratings     In stock  
2   3.9 out of 5 stars  14,699 ratings     In stock  
3   4.1 out of 5 stars   3,977 ratings     In stock  
4   4.1 out of 5 stars   3,244 ratings     In stock  
5   3.9 out of 5 stars   9,880 ratings     In stock  
6   4.0 out of 5 stars   9,417 ratings     In stock  
7   3.9 out of 5 stars   5,314 ratings     In stock  
8   4.1 out of 5 stars   3,244 ratings     In stock  
9   3.9 out of 5 stars   6,395 ratings     In stock  
10  4.2 out of 5 stars   5,512 ratings     In stock  
11  4.0 out of 5 stars  10,458 ratings     In stock  
12  4.0 out of 5 stars   9,417 ratings     In stock  
13  4.1 out of 5 stars   3,244 ratings     In stock  
14  4.0 out of 5 stars   7,716 ratings     In stock  
15       Previous page             NaN     In stock  
16  4.1 out of 5 stars  18,175 ratings     In stock  
17  3.9 out of 5 stars     123 ratings          NaN  
18  3.7 out of 5 stars     489 ratings     In stock  
19  4.1 out of 5 stars  18,175 ratings     In stock  
20  3.9 out of 5 stars     312 ratings     In stock  
21  3.8 out of 5 stars      20 ratings     In stock

In [ ]:
df.shape

(22, 5)

In [ ]:
# Drop rows with NaN values
df = df.dropna()

In [ ]:
df

title    price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32,999.   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15,990.   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8,499.   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9,899.   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14,499.   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11,999.   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10,999.   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5,299.   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12,999.   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14,999.   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25,999.   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6,799.   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10,999.   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12,999.   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7,299.   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12,999.   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12,599.   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12,999.   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11,999.   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19,999.   

                rating         reviews availability  
0   4.3 out of 5 stars   1,095 ratings     In stock  
1   4.0 out of 5 stars  12,391 ratings     In stock  
2   3.9 out of 5 stars  14,699 ratings     In stock  
3   4.1 out of 5 stars   3,977 ratings     In stock  
4   4.1 out of 5 stars   3,244 ratings     In stock  
5   3.9 out of 5 stars   9,880 ratings     In stock  
6   4.0 out of 5 stars   9,417 ratings     In stock  
7   3.9 out of 5 stars   5,314 ratings     In stock  
8   4.1 out of 5 stars   3,244 ratings     In stock  
9   3.9 out of 5 stars   6,395 ratings     In stock  
10  4.2 out of 5 stars   5,512 ratings     In stock  
11  4.0 out of 5 stars  10,458 ratings     In stock  
12  4.0 out of 5 stars   9,417 ratings     In stock  
13  4.1 out of 5 stars   3,244 ratings     In stock  
14  4.0 out of 5 stars   7,716 ratings     In stock  
16  4.1 out of 5 stars  18,175 ratings     In stock  
18  3.7 out of 5 stars     489 ratings     In stock  
19  4.1 out of 5 stars  18,175 ratings     In stock  
20  3.9 out of 5 stars     312 ratings     In stock  
21  3.8 out of 5 stars      20 ratings     In stock

# Question

## Q1: How many mobile phones are available under Rs. 16,500 that have a RAM capacity of at least 5GB?

In [ ]:
df['title']

0     iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...
1     Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...
2     Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...
3     Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...
4     realme narzo 60X 5G（Nebula Purple 6GB,128GB St...
5     realme narzo N53 (Feather Black, 8GB+128GB) 33...
6     realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...
7           Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)
8     realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...
9        Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM
10    iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...
11    Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...
12    realme narzo N55 (Prime Black, 6GB+128GB) 33W ...
13    realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...
14    realme narzo 50i Prime (Mint Green 4GB RAM+64G...
16    iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...
18    Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...
19    iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 1

In [ ]:
type(df['title'])

pandas.core.series.Series

In [ ]:
import re
def fun(x):
    return re.search(r'(\d)+GB', x).group(1)

ram=df['title'].apply(lambda x:fun(x))
ram

0     8
1     6
2     4
3     8
4     6
5     8
6     6
7     2
8     4
9     8
10    8
11    4
12    6
13    4
14    4
16    6
18    1
19    6
20    6
21    8
Name: title, dtype: object

In [ ]:
df.loc[:, 'RAM(in_GB)'] = ram

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\2831261713.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[:, 'RAM(in_GB)'] = ram


In [ ]:
df

title    price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32,999.   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15,990.   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8,499.   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9,899.   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14,499.   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11,999.   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10,999.   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5,299.   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12,999.   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14,999.   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25,999.   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6,799.   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10,999.   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12,999.   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7,299.   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12,999.   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12,599.   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12,999.   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11,999.   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19,999.   

                rating         reviews availability RAM(in_GB)  
0   4.3 out of 5 stars   1,095 ratings     In stock          8  
1   4.0 out of 5 stars  12,391 ratings     In stock          6  
2   3.9 out of 5 stars  14,699 ratings     In stock          4  
3   4.1 out of 5 stars   3,977 ratings     In stock          8  
4   4.1 out of 5 stars   3,244 ratings     In stock          6  
5   3.9 out of 5 stars   9,880 ratings     In stock          8  
6   4.0 out of 5 stars   9,417 ratings     In stock          6  
7   3.9 out of 5 stars   5,314 ratings     In stock          2  
8   4.1 out of 5 stars   3,244 ratings     In stock          4  
9   3.9 out of 5 stars   6,395 ratings     In stock          8  
10  4.2 out of 5 stars   5,512 ratings     In stock          8  
11  4.0 out of 5 stars  10,458 ratings     In stock          4  
12  4.0 out of 5 stars   9,417 ratings     In stock          6  
13  4.1 out of 5 stars   3,244 ratings     In stock          4  
14  4.0 out of 5 stars   7,716 ratings     In stock          4  
16  4.1 out of 5 stars  18,175 ratings     In stock          6  
18  3.7 out of 5 stars     489 ratings     In stock          1  
19  4.1 out of 5 stars  18,175 ratings     In stock          6  
20  3.9 out of 5 stars     312 ratings     In stock          6  
21  3.8 out of 5 stars      20 ratings     In stock          8

In [ ]:
df[['price', 'RAM(in_GB)']]


price RAM(in_GB)
0   32,999.          8
1   15,990.          6
2    8,499.          4
3    9,899.          8
4   14,499.          6
5   11,999.          8
6   10,999.          6
7    5,299.          2
8   12,999.          4
9   14,999.          8
10  25,999.          8
11   6,799.          4
12  10,999.          6
13  12,999.          4
14   7,299.          4
16  12,999.          6
18  12,599.          1
19  12,999.          6
20  11,999.          6
21  19,999.          8

In [ ]:
type(df.loc[1,'price'])

str

In [ ]:
df.loc[:,'price']

0     32,999.
1     15,990.
2      8,499.
3      9,899.
4     14,499.
5     11,999.
6     10,999.
7      5,299.
8     12,999.
9     14,999.
10    25,999.
11     6,799.
12    10,999.
13    12,999.
14     7,299.
16    12,999.
18    12,599.
19    12,999.
20    11,999.
21    19,999.
Name: price, dtype: object

In [ ]:
df.loc[:,'price']=df.loc[:,'price'].str.replace(',', '').str.replace('.', '')

In [ ]:
df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB)  
0   4.3 out of 5 stars   1,095 ratings     In stock          8  
1   4.0 out of 5 stars  12,391 ratings     In stock          6  
2   3.9 out of 5 stars  14,699 ratings     In stock          4  
3   4.1 out of 5 stars   3,977 ratings     In stock          8  
4   4.1 out of 5 stars   3,244 ratings     In stock          6  
5   3.9 out of 5 stars   9,880 ratings     In stock          8  
6   4.0 out of 5 stars   9,417 ratings     In stock          6  
7   3.9 out of 5 stars   5,314 ratings     In stock          2  
8   4.1 out of 5 stars   3,244 ratings     In stock          4  
9   3.9 out of 5 stars   6,395 ratings     In stock          8  
10  4.2 out of 5 stars   5,512 ratings     In stock          8  
11  4.0 out of 5 stars  10,458 ratings     In stock          4  
12  4.0 out of 5 stars   9,417 ratings     In stock          6  
13  4.1 out of 5 stars   3,244 ratings     In stock          4  
14  4.0 out of 5 stars   7,716 ratings     In stock          4  
16  4.1 out of 5 stars  18,175 ratings     In stock          6  
18  3.7 out of 5 stars     489 ratings     In stock          1  
19  4.1 out of 5 stars  18,175 ratings     In stock          6  
20  3.9 out of 5 stars     312 ratings     In stock          6  
21  3.8 out of 5 stars      20 ratings     In stock          8

In [ ]:
df.loc[:,'price']=df.loc[:,'price'].astype(int) # if not make equal to it with df[:,'price'] then not chages done in df then we can't use as int value

In [ ]:
df[df['price'] <= 16500]

title  price  \
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   

                rating         reviews availability RAM(in_GB)  
1   4.0 out of 5 stars  12,391 ratings     In stock          6  
2   3.9 out of 5 stars  14,699 ratings     In stock          4  
3   4.1 out of 5 stars   3,977 ratings     In stock          8  
4   4.1 out of 5 stars   3,244 ratings     In stock          6  
5   3.9 out of 5 stars   9,880 ratings     In stock          8  
6   4.0 out of 5 stars   9,417 ratings     In stock          6  
7   3.9 out of 5 stars   5,314 ratings     In stock          2  
8   4.1 out of 5 stars   3,244 ratings     In stock          4  
9   3.9 out of 5 stars   6,395 ratings     In stock          8  
11  4.0 out of 5 stars  10,458 ratings     In stock          4  
12  4.0 out of 5 stars   9,417 ratings     In stock          6  
13  4.1 out of 5 stars   3,244 ratings     In stock          4  
14  4.0 out of 5 stars   7,716 ratings     In stock          4  
16  4.1 out of 5 stars  18,175 ratings     In stock          6  
18  3.7 out of 5 stars     489 ratings     In stock          1  
19  4.1 out of 5 stars  18,175 ratings     In stock          6  
20  3.9 out of 5 stars     312 ratings     In stock          6

In [ ]:
type(df['RAM(in_GB)'][1])

str

In [ ]:
df.loc[:,'RAM(in_GB)']=df.loc[:,'RAM(in_GB)'].astype(int)

In [ ]:
df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB)  
0   4.3 out of 5 stars   1,095 ratings     In stock          8  
1   4.0 out of 5 stars  12,391 ratings     In stock          6  
2   3.9 out of 5 stars  14,699 ratings     In stock          4  
3   4.1 out of 5 stars   3,977 ratings     In stock          8  
4   4.1 out of 5 stars   3,244 ratings     In stock          6  
5   3.9 out of 5 stars   9,880 ratings     In stock          8  
6   4.0 out of 5 stars   9,417 ratings     In stock          6  
7   3.9 out of 5 stars   5,314 ratings     In stock          2  
8   4.1 out of 5 stars   3,244 ratings     In stock          4  
9   3.9 out of 5 stars   6,395 ratings     In stock          8  
10  4.2 out of 5 stars   5,512 ratings     In stock          8  
11  4.0 out of 5 stars  10,458 ratings     In stock          4  
12  4.0 out of 5 stars   9,417 ratings     In stock          6  
13  4.1 out of 5 stars   3,244 ratings     In stock          4  
14  4.0 out of 5 stars   7,716 ratings     In stock          4  
16  4.1 out of 5 stars  18,175 ratings     In stock          6  
18  3.7 out of 5 stars     489 ratings     In stock          1  
19  4.1 out of 5 stars  18,175 ratings     In stock          6  
20  3.9 out of 5 stars     312 ratings     In stock          6  
21  3.8 out of 5 stars      20 ratings     In stock          8

In [ ]:
ans1_df =df[(df['price'] <= 16500) & (df['RAM(in_GB)'] >= 5)]

In [ ]:
ans1_df

title  price  \
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   

                rating         reviews availability RAM(in_GB)  
1   4.0 out of 5 stars  12,391 ratings     In stock          6  
3   4.1 out of 5 stars   3,977 ratings     In stock          8  
4   4.1 out of 5 stars   3,244 ratings     In stock          6  
5   3.9 out of 5 stars   9,880 ratings     In stock          8  
6   4.0 out of 5 stars   9,417 ratings     In stock          6  
9   3.9 out of 5 stars   6,395 ratings     In stock          8  
12  4.0 out of 5 stars   9,417 ratings     In stock          6  
16  4.1 out of 5 stars  18,175 ratings     In stock          6  
19  4.1 out of 5 stars  18,175 ratings     In stock          6  
20  3.9 out of 5 stars     312 ratings     In stock          6

In [ ]:
ans1_df.groupby('RAM(in_GB)').count()['price']

RAM(in_GB)
6    7
8    3
Name: price, dtype: int64

In [ ]:
len(ans1_df)

10

### Based on the information gathered from the previous analyses, we can conclude that there are a total of 10 mobile phones in the dataset with prices less than or equal to 16500 and a minimum RAM capacity of 5 GB. Out of these, 7 mobile phones have 6 GB of RAM, and the remaining 3 mobile phones have 8 GB of RAM.

## Q2: Which mobile phones can a user purchase with a budget of up to Rs. 20,000 while ensuring that the rating is 4 or higher out of 5?

In [ ]:
# to apply any operation in data of any column follow this

In [ ]:
type(df['rating'])

pandas.core.series.Series

In [ ]:
type(df['rating'][1])

str

In [ ]:
df.loc[:,'rating'].str.split()

0     [4.3, out, of, 5, stars]
1     [4.0, out, of, 5, stars]
2     [3.9, out, of, 5, stars]
3     [4.1, out, of, 5, stars]
4     [4.1, out, of, 5, stars]
5     [3.9, out, of, 5, stars]
6     [4.0, out, of, 5, stars]
7     [3.9, out, of, 5, stars]
8     [4.1, out, of, 5, stars]
9     [3.9, out, of, 5, stars]
10    [4.2, out, of, 5, stars]
11    [4.0, out, of, 5, stars]
12    [4.0, out, of, 5, stars]
13    [4.1, out, of, 5, stars]
14    [4.0, out, of, 5, stars]
16    [4.1, out, of, 5, stars]
18    [3.7, out, of, 5, stars]
19    [4.1, out, of, 5, stars]
20    [3.9, out, of, 5, stars]
21    [3.8, out, of, 5, stars]
Name: rating, dtype: object

In [ ]:
df.loc[:,'rating'].str.split()[0][0]

'4.3'

In [ ]:
type(df.loc[:,'rating'].str.split()[0][0])

str

In [ ]:
ans2_df=df.loc[:,'rating'].str.split()

In [ ]:
ans2_df

0     [4.3, out, of, 5, stars]
1     [4.0, out, of, 5, stars]
2     [3.9, out, of, 5, stars]
3     [4.1, out, of, 5, stars]
4     [4.1, out, of, 5, stars]
5     [3.9, out, of, 5, stars]
6     [4.0, out, of, 5, stars]
7     [3.9, out, of, 5, stars]
8     [4.1, out, of, 5, stars]
9     [3.9, out, of, 5, stars]
10    [4.2, out, of, 5, stars]
11    [4.0, out, of, 5, stars]
12    [4.0, out, of, 5, stars]
13    [4.1, out, of, 5, stars]
14    [4.0, out, of, 5, stars]
16    [4.1, out, of, 5, stars]
18    [3.7, out, of, 5, stars]
19    [4.1, out, of, 5, stars]
20    [3.9, out, of, 5, stars]
21    [3.8, out, of, 5, stars]
Name: rating, dtype: object

In [ ]:
ans2_df.str.get(0)

0     4.3
1     4.0
2     3.9
3     4.1
4     4.1
5     3.9
6     4.0
7     3.9
8     4.1
9     3.9
10    4.2
11    4.0
12    4.0
13    4.1
14    4.0
16    4.1
18    3.7
19    4.1
20    3.9
21    3.8
Name: rating, dtype: object

In [ ]:
df.loc[:,'rating_value']=ans2_df.str.get(0) # this is done so make the above changes in dataframe

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\2630466384.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[:,'rating_value']=ans2_df.str.get(0) # this is done so make the above changes in dataframe


In [ ]:
df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB) rating_value  
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3  
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0  
2   3.9 out of 5 stars  14,699 ratings     In stock          4          3.9  
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1  
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1  
5   3.9 out of 5 stars   9,880 ratings     In stock          8          3.9  
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0  
7   3.9 out of 5 stars   5,314 ratings     In stock          2          3.9  
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1  
9   3.9 out of 5 stars   6,395 ratings     In stock          8          3.9  
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2  
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0  
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0  
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1  
14  4.0 out of 5 stars   7,716 ratings     In stock          4          4.0  
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1  
18  3.7 out of 5 stars     489 ratings     In stock          1          3.7  
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1  
20  3.9 out of 5 stars     312 ratings     In stock          6          3.9  
21  3.8 out of 5 stars      20 ratings     In stock          8          3.8

In [ ]:
df.loc[:,'rating_value']=df['rating_value'].astype(float)

In [ ]:
type(df.loc[:,'rating_value'][1])

float

In [ ]:
ans2_final_df =df[(df['price'] <= 20000) & (df['rating_value'] >= 4)]

In [ ]:
ans2_final_df

title  price  \
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0  
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1  
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1  
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0  
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1  
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0  
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0  
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1  
14  4.0 out of 5 stars   7,716 ratings     In stock          4          4.0  
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1  
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1

In [ ]:
len(ans2_final_df)

11

### Based on the analysis conducted above, it can be concluded that users have a choice of 11 mobile phones if their budget is up to 20000, and they desire a rating of 4 or higher.

## Q3: How many mobile phones have received more than 9000 reviews with a rating of 4 or higher?

In [ ]:
type(df['reviews'])

pandas.core.series.Series

In [ ]:
type(df['reviews'][1])

str

In [ ]:
df.loc[:,'reviews'].str.split()

0      [1,095, ratings]
1     [12,391, ratings]
2     [14,699, ratings]
3      [3,977, ratings]
4      [3,244, ratings]
5      [9,880, ratings]
6      [9,417, ratings]
7      [5,314, ratings]
8      [3,244, ratings]
9      [6,395, ratings]
10     [5,512, ratings]
11    [10,458, ratings]
12     [9,417, ratings]
13     [3,244, ratings]
14     [7,716, ratings]
16    [18,175, ratings]
18       [489, ratings]
19    [18,175, ratings]
20       [312, ratings]
21        [20, ratings]
Name: reviews, dtype: object

In [ ]:
ans3_df=df.loc[:,'reviews'].str.split()

In [ ]:
ans3_df.str.get(0)

0      1,095
1     12,391
2     14,699
3      3,977
4      3,244
5      9,880
6      9,417
7      5,314
8      3,244
9      6,395
10     5,512
11    10,458
12     9,417
13     3,244
14     7,716
16    18,175
18       489
19    18,175
20       312
21        20
Name: reviews, dtype: object

In [ ]:
df.loc[:,'no_of_reviews']=ans3_df.str.get(0)

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\1794449389.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[:,'no_of_reviews']=ans3_df.str.get(0)


In [ ]:
df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0   
2   3.9 out of 5 stars  14,699 ratings     In stock          4          3.9   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
5   3.9 out of 5 stars   9,880 ratings     In stock          8          3.9   
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
7   3.9 out of 5 stars   5,314 ratings     In stock          2          3.9   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
9   3.9 out of 5 stars   6,395 ratings     In stock          8          3.9   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0   
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
14  4.0 out of 5 stars   7,716 ratings     In stock          4          4.0   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
18  3.7 out of 5 stars     489 ratings     In stock          1          3.7   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
20  3.9 out of 5 stars     312 ratings     In stock          6          3.9   
21  3.8 out of 5 stars      20 ratings     In stock          8          3.8   

   no_of_reviews  
0          1,095  
1         12,391  
2         14,699  
3          3,977  
4          3,244  
5          9,880  
6          9,417  
7          5,314  
8          3,244  
9          6,395  
10         5,512  
11        10,458  
12         9,417  
13         3,244  
14         7,716  
16        18,175  
18           489  
19        18,175  
20           312  
21            20

In [ ]:
type(df['no_of_reviews'][1])

str

In [ ]:
df['no_of_reviews'].str.replace(",","")

0      1095
1     12391
2     14699
3      3977
4      3244
5      9880
6      9417
7      5314
8      3244
9      6395
10     5512
11    10458
12     9417
13     3244
14     7716
16    18175
18      489
19    18175
20      312
21       20
Name: no_of_reviews, dtype: object

In [ ]:
df.loc[:,'no_of_reviews']=df['no_of_reviews'].str.replace(",","")

In [ ]:
df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0   
2   3.9 out of 5 stars  14,699 ratings     In stock          4          3.9   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
5   3.9 out of 5 stars   9,880 ratings     In stock          8          3.9   
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
7   3.9 out of 5 stars   5,314 ratings     In stock          2          3.9   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
9   3.9 out of 5 stars   6,395 ratings     In stock          8          3.9   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0   
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
14  4.0 out of 5 stars   7,716 ratings     In stock          4          4.0   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
18  3.7 out of 5 stars     489 ratings     In stock          1          3.7   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
20  3.9 out of 5 stars     312 ratings     In stock          6          3.9   
21  3.8 out of 5 stars      20 ratings     In stock          8          3.8   

   no_of_reviews  
0           1095  
1          12391  
2          14699  
3           3977  
4           3244  
5           9880  
6           9417  
7           5314  
8           3244  
9           6395  
10          5512  
11         10458  
12          9417  
13          3244  
14          7716  
16         18175  
18           489  
19         18175  
20           312  
21            20

In [ ]:
df['no_of_reviews'].astype(int) # this will not change the datatype in dataframe

0      1095
1     12391
2     14699
3      3977
4      3244
5      9880
6      9417
7      5314
8      3244
9      6395
10     5512
11    10458
12     9417
13     3244
14     7716
16    18175
18      489
19    18175
20      312
21       20
Name: no_of_reviews, dtype: int32

In [ ]:
type(df['no_of_reviews'][5])

str

In [ ]:
df['no_of_reviews']=df['no_of_reviews'].astype(int)

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\307765301.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['no_of_reviews']=df['no_of_reviews'].astype(int)


In [ ]:
type(df['no_of_reviews'][5])

numpy.int32

In [ ]:
ans3_df =df[(df['no_of_reviews'] >= 9000) & (df['rating_value'] >= 4)]

In [ ]:
ans3_df

title  price  \
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0   
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0   
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews  
1           12391  
6            9417  
11          10458  
12           9417  
16          18175  
19          18175

In [ ]:
len(ans3_df)

6

### Based on the analysis conducted, it can be determined that a total of 6 mobile phones in the dataset have received reviews exceeding 9000 and have achieved a rating of 4 or more out of 5.

## Q4. I am a consumer who has a budget 20k. I wish to buy a phone taking in consideration the ratings and reviews of the phone . I want to spend the as less as possible but that does not mean that I want to buy the cheapest phone . Which phone should I buy ?


In [ ]:
ans4_df=df[df.loc[:,'price']<=20000]

In [ ]:
ans4_df

title  price  \
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB) rating_value  \
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0   
2   3.9 out of 5 stars  14,699 ratings     In stock          4          3.9   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
5   3.9 out of 5 stars   9,880 ratings     In stock          8          3.9   
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
7   3.9 out of 5 stars   5,314 ratings     In stock          2          3.9   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
9   3.9 out of 5 stars   6,395 ratings     In stock          8          3.9   
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0   
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
14  4.0 out of 5 stars   7,716 ratings     In stock          4          4.0   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
18  3.7 out of 5 stars     489 ratings     In stock          1          3.7   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
20  3.9 out of 5 stars     312 ratings     In stock          6          3.9   
21  3.8 out of 5 stars      20 ratings     In stock          8          3.8   

    no_of_reviews  
1           12391  
2           14699  
3            3977  
4            3244  
5            9880  
6            9417  
7            5314  
8            3244  
9            6395  
11          10458  
12           9417  
13           3244  
14           7716  
16          18175  
18            489  
19          18175  
20            312  
21             20

We're only looking at mobile phones that have a rating of higher than 4. If a phone has a rating below or equal to 4, we don't consider it as a good mobile.

In [ ]:
ans4_df=ans4_df[(ans4_df['rating_value'] > 4)]

Now, it may be possible that there were mobile phones that had a higher rating but a lower number of reviews. We will also not consider those mobile phones, so for this, we calculate ratings and reviews for each row, and then we consider only those rows or mobiles that have a lower ratio of ratings and reviews.

In [ ]:
ans4_df.loc[:, 'rating/reviews'] = (ans4_df['rating_value'] / ans4_df['no_of_reviews'])*10000 # this multiplication only to get good looking values

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\2210509369.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ans4_df.loc[:, 'rating/reviews'] = (ans4_df['rating_value'] / ans4_df['no_of_reviews'])*10000 # this multiplication only to get good looking values


In [ ]:
ans4_df

title  price  \
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews rating/reviews  
3            3977      10.309278  
4            3244      12.638718  
8            3244      12.638718  
13           3244      12.638718  
16          18175       2.255846  
19          18175       2.255846

In [ ]:
ans4_df = ans4_df.sort_values(by='rating/reviews', ascending=False)

In [ ]:
ans4_df

title  price  \
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews rating/reviews  
4            3244      12.638718  
8            3244      12.638718  
13           3244      12.638718  
3            3977      10.309278  
16          18175       2.255846  
19          18175       2.255846

Hence, we conclude that if consumers have to spend as less as possible, then they have to buy a "Xiaomi Redmi 10 Power (Power Black, 8GB RAM)" mobile with a price of only "9899" with a good number of reviews and a good rating too.

## Q5.now if my budget is 50k , which phone do I buy  with same scenario as of Q4??

In [ ]:
ans5_df=df[df.loc[:,'price']<=50000]

In [ ]:
ans5_df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
1   Samsung Galaxy M14 5G (ICY Silver,6GB,128GB)|5...  15990   
2   Samsung Galaxy M04 Light Green, 4GB RAM, 64GB ...   8499   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
5   realme narzo N53 (Feather Black, 8GB+128GB) 33...  11999   
6   realme narzo N55 (Prime Blue, 6GB+128GB) 33W S...  10999   
7         Redmi A2 (Sea Green, 2GB RAM, 64GB Storage)   5299   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
9      Redmi 12 5G Moonstone Silver 8GB RAM 256GB ROM  14999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
11  Redmi 12C (Matte Black, 4GB RAM, 64GB Storage)...   6799   
12  realme narzo N55 (Prime Black, 6GB+128GB) 33W ...  10999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
14  realme narzo 50i Prime (Mint Green 4GB RAM+64G...   7299   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
18  Nokia G42 5G | Snapdragon® 480+ 5G | 50MP Trip...  12599   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
20  Redmi Note 12 (Sunrise Gold, 6GB RAM, 64GB Sto...  11999   
21  Oppo A79 5G (Glowing Green, 8GB RAM, 128GB Sto...  19999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
1   4.0 out of 5 stars  12,391 ratings     In stock          6          4.0   
2   3.9 out of 5 stars  14,699 ratings     In stock          4          3.9   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
5   3.9 out of 5 stars   9,880 ratings     In stock          8          3.9   
6   4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
7   3.9 out of 5 stars   5,314 ratings     In stock          2          3.9   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
9   3.9 out of 5 stars   6,395 ratings     In stock          8          3.9   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
11  4.0 out of 5 stars  10,458 ratings     In stock          4          4.0   
12  4.0 out of 5 stars   9,417 ratings     In stock          6          4.0   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
14  4.0 out of 5 stars   7,716 ratings     In stock          4          4.0   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
18  3.7 out of 5 stars     489 ratings     In stock          1          3.7   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
20  3.9 out of 5 stars     312 ratings     In stock          6          3.9   
21  3.8 out of 5 stars      20 ratings     In stock          8          3.8   

    no_of_reviews  
0            1095  
1           12391  
2           14699  
3            3977  
4            3244  
5            9880  
6            9417  
7            5314  
8            3244  
9            6395  
10           5512  
11          10458  
12           9417  
13           3244  
14           7716  
16          18175  
18            489  
19          18175  
20            312  
21             20

We're only looking at mobile phones that have a rating  higher than 4. If a phone has a rating below or equal to 4, we don't consider it as a good mobile.

In [ ]:
ans5_df=ans5_df[(ans5_df['rating_value'] > 4)]

In [ ]:
ans5_df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews  
0            1095  
3            3977  
4            3244  
8            3244  
10           5512  
13           3244  
16          18175  
19          18175

Now, it may be possible that there were mobile phones that had a higher rating but a lower number of reviews. We will also not consider those mobile phones, so for this, we calculate ratings and reviews for each row, and then we consider only those rows or mobiles that have a lower ratio of ratings and reviews.

In [ ]:
ans5_df.loc[:, 'rating/reviews'] = (ans5_df['rating_value'] / ans5_df['no_of_reviews'])*10000 # this multiplication only to get good looking values

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\439990011.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ans5_df.loc[:, 'rating/reviews'] = (ans5_df['rating_value'] / ans5_df['no_of_reviews'])*10000 # this multiplication only to get good looking values


In [ ]:
ans5_df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews rating/reviews  
0            1095      39.269406  
3            3977      10.309278  
4            3244      12.638718  
8            3244      12.638718  
10           5512       7.619739  
13           3244      12.638718  
16          18175       2.255846  
19          18175       2.255846

In [ ]:
ans5_df = ans5_df.sort_values(by='rating/reviews', ascending=False)

In [ ]:
ans5_df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews rating/reviews  
0            1095      39.269406  
4            3244      12.638718  
8            3244      12.638718  
13           3244      12.638718  
3            3977      10.309278  
10           5512       7.619739  
16          18175       2.255846  
19          18175       2.255846

Hence, we conclude that if consumers have to spend as less as possible, then they have to buy a "Xiaomi Redmi 10 Power (Power Black, 8GB RAM)" mobile with a price of only "9899" and a good number of reviews and a good rating too. But as the consumer has a budget of 50,000, there is a mobile he can buy, which is approximately 1/4 of his budget: "iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)" or "iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)."

## Q6. Now I am a shopkeeper , who wishes to do some analysis to buy phones for my shop based on the data provided . Which phone do I buy to ensure maximum sales in my shop ??


In [ ]:
df['rating_value'].min()

3.7

In [ ]:
df['rating_value'].max()

4.3

For the maximum sale we choose the mobile which are getting good rating and have good number of reviews means first filter the data by rating > 4 and then in that we filter it by more number of reviews means less rating/reviews ratio

In [ ]:
ans6_df=df[(df['rating_value'] > 4)]

In [ ]:
ans6_df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews  
0            1095  
3            3977  
4            3244  
8            3244  
10           5512  
13           3244  
16          18175  
19          18175

In [ ]:
ans6_df.loc[:, 'rating/reviews'] = (ans6_df['rating_value'] / ans6_df['no_of_reviews'])*10000 # this multiplication only to get good looking values

C:\Users\ROHIT MOTWANI\AppData\Local\Temp\ipykernel_17876\2995148861.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ans6_df.loc[:, 'rating/reviews'] = (ans6_df['rating_value'] / ans6_df['no_of_reviews'])*10000 # this multiplication only to get good looking values


In [ ]:
ans6_df

title  price  \
0   iQOO Neo 7 Pro 5G (Dark Storm, 8GB RAM, 128GB ...  32999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
0   4.3 out of 5 stars   1,095 ratings     In stock          8          4.3   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews rating/reviews  
0            1095      39.269406  
3            3977      10.309278  
4            3244      12.638718  
8            3244      12.638718  
10           5512       7.619739  
13           3244      12.638718  
16          18175       2.255846  
19          18175       2.255846

And we will be considering those mobile whose rating/review ratio is less equal to 15 for getting the good sale

In [ ]:
ans6_df=ans6_df[(ans6_df['rating/reviews'] <= 15)]

In [ ]:
ans6_df

title  price  \
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   

    no_of_reviews rating/reviews  
3            3977      10.309278  
4            3244      12.638718  
8            3244      12.638718  
10           5512       7.619739  
13           3244      12.638718  
16          18175       2.255846  
19          18175       2.255846

### If the shopkeeper wants to maximize sales and, consequently, profits, the suggested list of mobiles to purchase includes the following models:

- iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)
- iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)
- realme narzo 60X 5G（Stellar Green, 4GB, 128GB)
- iQOO Neo 7 5G (Frost Blue, 8GB RAM, 12realme
- narzo 60X 5G（Nebula Purple 4GB, 128GB)
- realme narzo 60X 5G（Nebula Purple 6GB,128GB)
- Xiaomi Redmi 10 Power (Power Black, 8GB RAM)

#### These models are recommended for their potential to attract customers and drive sales, leading to higher profitability.

## Q7.what quantity of this model of phones do I buy ??

In [ ]:
ans6_df = ans6_df.sort_values(by='rating/reviews', ascending=True)

In [ ]:
ans6_df

title  price  \
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   

    no_of_reviews rating/reviews  
16          18175       2.255846  
19          18175       2.255846  
10           5512       7.619739  
3            3977      10.309278  
4            3244      12.638718  
8            3244      12.638718  
13           3244      12.638718

The priority order according to which the shopkeeper can decide the quantity of  mobile phones he have to buy is as follows:

- iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)
- iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)
- iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage)
- Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB)
- realme narzo 60X 5G (Nebula Purple 6GB,128GB)
- realme narzo 60X 5G (Nebula Purple 4GB, 128GB)
- realme narzo 60X 5G (Stellar Green, 4GB, 128GB)

## Q8.Now suppose I have a budget of 200k and I can buy more than one model of phones , instead of just one model . What do I do ? Do I still buy the same model's phones or different phones ? Give me the ideal distribution of models of phones I should buy for maximum sales .

In [ ]:
ans6_df

title  price  \
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   

                rating         reviews availability RAM(in_GB) rating_value  \
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   

    no_of_reviews rating/reviews  
16          18175       2.255846  
19          18175       2.255846  
10           5512       7.619739  
3            3977      10.309278  
4            3244      12.638718  
8            3244      12.638718  
13           3244      12.638718

To buy the maximum mobile with limited buget the list of mobile should be according to price first

In [ ]:
ans6_df = ans6_df.sort_values(by='price', ascending=True)

In [ ]:
ans6_df

title  price  \
3   Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...   9899   
16  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...  12999   
19  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...  12999   
8   realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...  12999   
13  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...  12999   
4   realme narzo 60X 5G（Nebula Purple 6GB,128GB St...  14499   
10  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...  25999   

                rating         reviews availability RAM(in_GB) rating_value  \
3   4.1 out of 5 stars   3,977 ratings     In stock          8          4.1   
16  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
19  4.1 out of 5 stars  18,175 ratings     In stock          6          4.1   
8   4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
13  4.1 out of 5 stars   3,244 ratings     In stock          4          4.1   
4   4.1 out of 5 stars   3,244 ratings     In stock          6          4.1   
10  4.2 out of 5 stars   5,512 ratings     In stock          8          4.2   

    no_of_reviews rating/reviews  
3            3977      10.309278  
16          18175       2.255846  
19          18175       2.255846  
8            3244      12.638718  
13           3244      12.638718  
4            3244      12.638718  
10           5512       7.619739

In [ ]:
order_list = []
amount = 200000
min_amt=ans6_df['price'].min()
while(amount >= min_amt):
    for index, row in ans6_df.iterrows():
        price = row['price']

        if amount != 0 and price <= amount:
            order_list.append(row['title'])
            amount -= price
        else:
            break

order_list

['Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB Storage)',
 "iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB Storage) with Charger | World's First Snapdragon 4 Gen 1 | Best in-Segment 120Hz Refresh Rate | Travel Adaptor Included in The Box",
 "iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB Storage) with Charger | World's First Snapdragon 4 Gen 1 | Best in-Segment 120Hz Refresh Rate | Travel Adaptor Included in The Box",
 'realme narzo 60X 5G（Nebula Purple 4GB, 128GB Storage） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge',
 'realme narzo 60X 5G（Stellar Green, 4GB, 128GB Storage） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge',
 'realme narzo 60X 5G（Nebula Purple 6GB,128GB Storage ） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge',
 'iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage) | Dimensity 8200, only 4nm Processor in The Segment | 50% Charge in 10 mins | 

In [ ]:
amount

1415

In [ ]:
# Convert to a set to get unique values
unique_values = set(order_list)

# Convert back to a list
unique_order_list = list(unique_values)

In [ ]:
unique_order_list

['realme narzo 60X 5G（Stellar Green, 4GB, 128GB Storage） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge',
 "iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB Storage) with Charger | World's First Snapdragon 4 Gen 1 | Best in-Segment 120Hz Refresh Rate | Travel Adaptor Included in The Box",
 'iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage) | Dimensity 8200, only 4nm Processor in The Segment | 50% Charge in 10 mins | Motion Control & 90 FPS Gaming',
 'Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB Storage)',
 'realme narzo 60X 5G（Nebula Purple 4GB, 128GB Storage） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge',
 'realme narzo 60X 5G（Nebula Purple 6GB,128GB Storage ） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge',
 "iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB Storage) with Charger | World's First Snapdragon 4 Gen 1 | Best in-Segment 120Hz Refresh Rate | Trav

In [ ]:
from collections import Counter
counter=Counter(order_list)


In [ ]:
counter

Counter({'Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB Storage)': 4,
         "iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB Storage) with Charger | World's First Snapdragon 4 Gen 1 | Best in-Segment 120Hz Refresh Rate | Travel Adaptor Included in The Box": 2,
         "iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB Storage) with Charger | World's First Snapdragon 4 Gen 1 | Best in-Segment 120Hz Refresh Rate | Travel Adaptor Included in The Box": 2,
         'realme narzo 60X 5G（Nebula Purple 4GB, 128GB Storage） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge': 2,
         'realme narzo 60X 5G（Stellar Green, 4GB, 128GB Storage） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge': 2,
         'realme narzo 60X 5G（Nebula Purple 6GB,128GB Storage ） Up to 2TB External Memory | 50 MP AI Primary Camera | Segments only 33W Supervooc Charge': 2,
         'iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage) | Di

In [ ]:
ans8_df = pd.DataFrame(list(counter.items()), columns=['Title', 'No._of_Products'])

In [ ]:
ans8_df

Title  No._of_Products
0  Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 1...                4
1  iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB ...                2
2  iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB...                2
3  realme narzo 60X 5G（Nebula Purple 4GB, 128GB S...                2
4  realme narzo 60X 5G（Stellar Green, 4GB, 128GB ...                2
5  realme narzo 60X 5G（Nebula Purple 6GB,128GB St...                2
6  iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Stor...                1

### Hence, this is the list of mobiles with the quatity the shopkeeper has  to buy for their shop with a capital of 200k


| Title                                               | No._of_Products |
|-----------------------------------------------------|-----------------|
| Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB) | 4               |
| iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)      | 2               |
| iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)     | 2               |
| realme narzo 60X 5G（Nebula Purple 4GB, 128GB)      | 2               |
| realme narzo 60X 5G（Stellar Green, 4GB, 128GB)     | 2               |
| realme narzo 60X 5G（Nebula Purple 6GB,128GB)       | 2               |
| iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage)  | 1               |



# Conclusion

### Q1: How many mobile phones are available under Rs. 16,500 that have a RAM capacity of at least 5GB?

 Based on the information gathered from the previous analyses, we can conclude that there are a total of 10 mobile phones in the dataset with prices less than or equal to 16500 and a minimum RAM capacity of 5 GB. Out of these, 7 mobile phones have 6 GB of RAM, and the remaining 3 mobile phones have 8 GB of RAM.

### Q2: Which mobile phones can a user purchase with a budget of up to Rs. 20,000 while ensuring that the rating is 4 or higher out of 5?

 Based on the analysis conducted above, it can be concluded that users have a choice of 11 mobile phones if their budget is up to 20000, and they desire a rating of 4 or higher.

### Q3: How many mobile phones have received more than 9000 reviews with a rating of 4 or higher?

 Based on the analysis conducted, it can be determined that a total of 6 mobile phones in the dataset have received reviews exceeding 9000 and have achieved a rating of 4 or more out of 5.

### Q4. I am a consumer who has a budget 20k. I wish to buy a phone taking in consideration the ratings and reviews of the phone . I want to spend the as less as possible but that does not mean that I want to buy the cheapest phone . Which phone should I buy ?

Hence, we conclude that if consumers have to spend as less as possible, then they have to buy a "Xiaomi Redmi 10 Power (Power Black, 8GB RAM)" mobile with a price of only "9899" with a good number of reviews and a good rating too.

### Q5.now if my budget is 50k , which phone do I buy with same scenario as of Q4??

Hence, we conclude that if consumers have to spend as less as possible, then they have to buy a "Xiaomi Redmi 10 Power (Power Black, 8GB RAM)" mobile with a price of only "9899" and a good number of reviews and a good rating too. But as the consumer has a budget of 50,000, there is a mobile he can buy, which is approximately 1/4 of his budget: "iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)" or "iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)."

### Q6. Now I am a shopkeeper , who wishes to do some analysis to buy phones for my shop based on the data provided . Which phone do I buy to ensure maximum sales in my shop ??

If the shopkeeper wants to maximize sales and, consequently, profits, the suggested list of mobiles to purchase includes the following models:
- iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)
- iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)
- realme narzo 60X 5G（Stellar Green, 4GB, 128GB)
- iQOO Neo 7 5G (Frost Blue, 8GB RAM, 12realme
- narzo 60X 5G（Nebula Purple 4GB, 128GB)
- realme narzo 60X 5G（Nebula Purple 6GB,128GB)
- Xiaomi Redmi 10 Power (Power Black, 8GB RAM)

These models are recommended for their potential to attract customers and drive sales, leading to higher profitability.

### Q7.what quantity of this model of phones do I buy ??

The priority order according to which the shopkeeper can decide the quantity of mobile phones he have to buy is as follows:

- iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)
- iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)
- iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage)
- Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB)
- realme narzo 60X 5G (Nebula Purple 6GB,128GB)
- realme narzo 60X 5G (Nebula Purple 4GB, 128GB)
- realme narzo 60X 5G (Stellar Green, 4GB, 128GB)

### Q8.Now suppose I have a budget of 200k and I can buy more than one model of phones , instead of just one model . What do I do ? Do I still buy the same model's phones or different phones ? Give me the ideal distribution of models of phones I should buy for maximum sales .

 Hence, this is the list of mobiles with the quatity the shopkeeper has  to buy for their shop with a capital of 200k


| Title                                               | No._of_Products |
|-----------------------------------------------------|-----------------|
| Xiaomi Redmi 10 Power (Power Black, 8GB RAM, 128GB) | 4               |
| iQOO Z6 Lite 5G (Mystic Night, 6GB RAM, 128GB)      | 2               |
| iQOO Z6 Lite 5G (Stellar Green, 6GB RAM, 128GB)     | 2               |
| realme narzo 60X 5G（Nebula Purple 4GB, 128GB)      | 2               |
| realme narzo 60X 5G（Stellar Green, 4GB, 128GB)     | 2               |
| realme narzo 60X 5G（Nebula Purple 6GB,128GB)       | 2               |
| iQOO Neo 7 5G (Frost Blue, 8GB RAM, 128GB Storage)  | 1               |



END